connectors/sources/mongo.py

```diff
@@ -246,28 +249,29 @@ def _serialize(value):
         return doc
 
     async def get_docs(self, filtering=None):
-        if filtering is not None and filtering.has_advanced_rules():
-            advanced_rules = filtering.get_advanced_rules()
+        with self.get_client() as client:
+            db = client[self.configuration["database"]]
+            collection = db[self.configuration["collection"]]
+
+            if filtering is not None and filtering.has_advanced_rules():
```

Comment:
Nit, we might be able to shorten this statement a bit:

```suggestion
            if filtering and filtering.has_advanced_rules():
```